src/main/java/com/jagrosh/jmusicbot/audio/QueuedTrack.java

```diff
@@ -28,33 +28,38 @@
 public class QueuedTrack implements Queueable
 {
     private final AudioTrack track;
-    
-    public QueuedTrack(AudioTrack track, User owner)
-    {
-        this(track, new RequestMetadata(owner));
-    }
-    
+    private final RequestMetadata requestMetadata;
+
     public QueuedTrack(AudioTrack track, RequestMetadata rm)
     {
         this.track = track;
         this.track.setUserData(rm == null ? RequestMetadata.EMPTY : rm);
+
+        this.requestMetadata = rm;
+        if (this.track.isSeekable() && rm != null)
+            track.setPosition(rm.requestInfo.startTimestamp);
     }
     
     @Override
     public long getIdentifier() 
     {
-        return track.getUserData() == null ? 0L : track.getUserData(RequestMetadata.class).getOwner();
+        return requestMetadata.getOwner();
```

Comment:
I unintentionally got rid of this null check while solving the merge conflicts, I'm not sure whether this has any significance so I'd need to double-check

Reply:
Looking at the implementation of `RequestMetadata#getOwner` it does the null check there as well, so this should be fine.

https://github.com/jagrosh/MusicBot/blob/915735da0333f327275b5621f0fc9c37f7e36299/src/main/java/com/jagrosh/jmusicbot/audio/RequestMetadata.java#L43-L46